ivy/utils/backend/sub_backend_handler.py

```diff
@@ -0,0 +1,187 @@
+import os
+from types import ModuleType, FunctionType
+import warnings
+
+import ivy
+from ivy.func_wrapper import _wrap_function
+from ivy.utils.exceptions import IvyException
+
+warnings.simplefilter("always", UserWarning)
+
+
+_backends_subpackage_path = "ivy.functional.backends"
+_sub_backend_dict = dict()
+_backend_to_sub_backends_dict = dict()
+
+
+# dynamic sub_backend detection
+for backend in os.listdir(
+    os.path.join(
+        ivy.__path__[0].rpartition(os.path.sep)[0],  # type: ignore
+        _backends_subpackage_path.replace(".", os.path.sep),
+    )
+):
+    if backend.startswith("__"):
+        continue
+
+    sub_backends_dir = os.path.join(
+        ivy.__path__[0].rpartition(os.path.sep)[0],
+        _backends_subpackage_path.replace(".", os.path.sep),
+        backend,
+        "sub_backends",
+    )
+    if os.path.isdir(sub_backends_dir):
+        for sub_backend in os.listdir(sub_backends_dir):
+            if sub_backend.startswith("__"):
+                continue
+            _sub_backend_dict[
+                sub_backend
+            ] = f"{_backends_subpackage_path}.{backend}.sub_backends.{sub_backend}"
+            try:
+                _backend_to_sub_backends_dict[backend].append(sub_backend)
+            except KeyError:
+                _backend_to_sub_backends_dict[backend] = [sub_backend]
+
+
+_all_sub_backends = []
+
+[_all_sub_backends.extend(v) for v in _backend_to_sub_backends_dict.values()]
```

Comment:
We're constructing a list here and throwing it away, let's use a normal for loop.